rules/naming/src/ExpectedNameResolver/AbstractExpectedNameResolver.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Naming\ExpectedNameResolver;
+
+use Nette\Utils\Strings;
+use PhpParser\Node\Stmt\Property;
+use Rector\Naming\Naming\PropertyNaming;
+use Rector\NodeNameResolver\NodeNameResolver;
+use Rector\NodeTypeResolver\NodeTypeResolver;
+
+abstract class AbstractExpectedNameResolver implements ExpectedNameResolverInterface
```

Comment:
For now AbstractExpectedNameResolver, but it is very possible that will be renamed to AbstractPropertyExpectedNameResolver when Param, Variable, etc will be refactored to new approach